integration-tests/testsetups/mercury.go

```diff
@@ -0,0 +1,279 @@
+	"github.com/smartcontractkit/chainlink/integration-tests/contracts"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/guregu/null.v4"
+)
+
+// type MercuryTest struct {
+// 	Namespace     string
+// 	IsExistingEnv bool
+// }
+
+// func NewMercuryTest() *MercuryTest {
+// 	existingNamespace := os.Getenv(cl_env_config.EnvVarNamespace)
+
+// 	return &MercuryTest{
+// 		Namespace:     existingNamespace,
+// 		IsExistingEnv: existingNamespace != "",
+// 	}
+// }
+
+func SetupMercuryEnv(t *testing.T) (
+	*environment.Environment, bool, blockchain.EVMNetwork, []*client.Chainlink, string,
+	blockchain.EVMClient, *ctfClient.MockserverClient, *client.MercuryServer) {
+	testNetwork := networks.SelectedNetwork
+	evmConfig := eth.New(nil)
+	if !testNetwork.Simulated {
+		evmConfig = eth.New(&eth.Props{
+			NetworkName: testNetwork.Name,
+			Simulated:   testNetwork.Simulated,
+			WsURLs:      testNetwork.URLs,
+		})
+	}
+
+	mercuryServerInternalUrl := fmt.Sprintf("http://%s:3000", mercury_server.URLsKey)
+	secretsToml := fmt.Sprintf("%s\n%s\n%s\n%s\n",
+		"[[Mercury.Credentials]]",
+		fmt.Sprintf(`URL = "%s/reports"`, mercuryServerInternalUrl),
+		`Username = "node"`,
+		`Password = "nodepass"`,
+	)
+
+	testEnvironment := environment.New(&environment.Config{
+		// TTL:             1 * time.Hour,
+		NamespacePrefix: fmt.Sprintf("smoke-mercury-%s", strings.ReplaceAll(strings.ToLower(testNetwork.Name), " ", "-")),
+		Test:            t,
+	}).
+		AddHelm(mockservercfg.New(nil)).
+		AddHelm(mockserver.New(nil)).
+		AddHelm(evmConfig).
+		AddHelm(mercury_server.New(map[string]interface{}{
+			"imageRepo": os.Getenv("MERCURY_SERVER_IMAGE"),
+			"imageTag":  os.Getenv("MERCURY_SERVER_TAG"),
+		})).
+		AddHelm(chainlink.New(0, map[string]interface{}{
+			"replicas": "5",
+			"toml": client.AddNetworksConfig(
+				config.BaseMercuryTomlConfig,
+				testNetwork),
+			"secretsToml": secretsToml,
+		}))
+	err := testEnvironment.Run()
+	require.NoError(t, err, "Error running test environment")
+
+	chainlinkNodes, err := client.ConnectChainlinkNodes(testEnvironment)
+	require.NoError(t, err, "Error connecting to Chainlink nodes")
+	require.NoError(t, err, "Retreiving on-chain wallet addresses for chainlink nodes shouldn't fail")
+
+	evmClient, err := blockchain.NewEVMClient(testNetwork, testEnvironment)
+	require.NoError(t, err, "Error connecting to blockchain")
+
+	isExistingTestEnv := os.Getenv(cl_env_config.EnvVarNamespace) != "" && os.Getenv(cl_env_config.EnvVarNoManifestUpdate) == "true"
+
+	// Setup random mock server response for mercury price feed
+	mockserverClient, err := ctfClient.ConnectMockServer(testEnvironment)
+	require.NoError(t, err, "Error connecting to mock server")
+
+	mercuryServerRemoteUrl := testEnvironment.URLs[mercury_server.URLsKey][1]
+	mercuryServerClient := client.NewMercuryServer(mercuryServerRemoteUrl)
+
+	t.Cleanup(func() {
```

Comment:
A good practice is to use `t.Cleanup` or require/assert packages only in your test, not in helpers.
When we are going to migrate to `chainlink-ethereum` I must refactor all that stuff to return `error` because interfaces won't contain `*testing.T`
By removing `*testing.T` from helpers you allow people to use your setup not only in tests.